server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -4237,6 +4246,274 @@ public VlanVO doInTransaction(final TransactionStatus status) {
+            final String currentEndIPv6 = existingVlanIPRangeArray[1];
+
+            if (startIpv6 == null) {
+                startIpv6 = currentStartIPv6;
+            }
+            if (endIpv6 == null) {
+                endIpv6 = currentEndIPv6;
+            }
+            if (startIpv6 != currentStartIPv6 || endIpv6 != currentEndIPv6) {
+                _networkModel.checkIp6Parameters(startIpv6, endIpv6, ip6Gateway, ip6Cidr);
+                final List<UserIpv6AddressVO> ips = _ipv6Dao.listByVlanId(id);
+                checkAllocatedIpv6sAreWithinVlanRange(ips, startIp, endIp);
+
+                try {
+                    VlanVO range = _vlanDao.acquireInLockTable(id, 30);
+                    if (range == null) {
+                        throw new CloudRuntimeException("Unable to acquire vlan configuration: " + id);
+                    }
+
+                    if (s_logger.isDebugEnabled()) {
+                        s_logger.debug("lock vlan " + id + " is acquired");
+                    }
+
+                    commitUpdateVlanAndIpRange(id, startIpv6, endIpv6, currentStartIPv6, currentEndIPv6, false, isRangeForSystemVM,forSystemVms);
+
+                } catch (final Exception e) {
+                    s_logger.error("Unable to edit VlanRange due to " + e.getMessage(), e);
+                    throw new CloudRuntimeException("Failed to edit VlanRange. Please contact Cloud Support.");
+                } finally {
+                    _vlanDao.releaseFromLockTable(id);
+                }
+            }
+        }
+
+        return _vlanDao.findById(id);
+    }
+
+    private VlanVO commitUpdateVlanAndIpRange(final Long id, final String newStartIP, final String newEndIP, final String currentStartIP, final String currentEndIP,
+                                              final boolean ipv4, final Boolean isRangeForSystemVM, final Boolean forSystemvms) {
+
+        return Transaction.execute(new TransactionCallback<VlanVO>() {
+            @Override
+            public VlanVO doInTransaction(final TransactionStatus status) {
+                VlanVO vlanRange = _vlanDao.findById(id);
+                s_logger.debug("Updating vlan range " + vlanRange.getId());
+                if (ipv4) {
+                    vlanRange.setIpRange(newStartIP + "-" + newEndIP);
+                    _vlanDao.update(vlanRange.getId(), vlanRange);
+                    if (!updatePublicIPRange(newStartIP, currentStartIP, newEndIP, currentEndIP, vlanRange.getDataCenterId(), vlanRange.getId(), vlanRange.getNetworkId(), vlanRange.getPhysicalNetworkId(), isRangeForSystemVM, forSystemvms)) {
+                        throw new CloudRuntimeException("Failed to update IPv4 range. Please contact Cloud Support.");
+                    }
+                } else {
+                    vlanRange.setIp6Range(newStartIP + "-" + newEndIP);
+                    _vlanDao.update(vlanRange.getId(), vlanRange);
```

Comment:
@weizhouapache update public IP range required here same as ipv4 above?

Reply:
@sureshanaparti 
it is not needed. 
there is a record in user_ip_address table for each ipv4 address. however, user_ipv6_address table only contains the ipv6 address which are in use (Allocated state).